core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -304,8 +316,8 @@ private SearcherManager createSearcherManager() throws EngineException {
                 throw new EngineCreationFailureException(shardId, "failed to open reader on writer", e);
             }
         } finally {
-            if (success == false) { // release everything we created on a failure
-                IOUtils.closeWhileHandlingException(searcherManager, indexWriter);
+            if (searcherManager == null) { // release everything we created on a failure
```

Comment:
I think we are actually OK here: the finally clause in the ctor would close the first searcherManager.  I agree it's confusing ...